internal/provider/provider.go

```diff
@@ -321,6 +361,31 @@ func (p *fluxProvider) Configure(ctx context.Context, req provider.ConfigureRequ
 	if data.Git.AuthorName.IsNull() {
 		data.Git.AuthorName = types.StringValue(defaultAuthor)
 	}
+	if data.Kubernetes.LoadConfigFile.IsNull() {
+		data.Kubernetes.LoadConfigFile = types.BoolValue(true)
+	}
+	if data.Kubernetes.ConfigPaths.IsNull() {
+		if v, ok := os.LookupEnv("KUBE_CONFIG_PATHS"); ok {
+			var paths []attr.Value
+			for _, p := range filepath.SplitList(v) {
+				paths = append(paths, types.StringValue(p))
+			}
+			data.Kubernetes.ConfigPaths = types.SetValueMust(types.StringType, paths)
+		}
+	}
+	if data.Kubernetes.ConfigPath.IsNull() {
+		data.Kubernetes.ConfigPath = types.StringValue("~/.kube/config")
+		defaultEnv := []string{
+			"KUBE_CONFIG",
```

Comment:
This was taken from the kubectl provider. I think it's a valid point and it can be removed. Alternatively, the helm provider uses a dedicated environment variable `KUBE_CONFIG_PATH` for this and does not rely on a default value.